File: Java/ProgramAdviser/app/src/main/java/bjfu/it/zhangsixuan/programadviser/MainActivity.java
package bjfu.it.zhangsixuan.programadviser;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Spinner;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    private final ProgramExpert programExpert = new ProgramExpert();
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
    public void onClickButton(View button) {

        // get spinner 引用
        Spinner spinner = findViewById(R.id.feature);

        //get spinner 选项
        String feature = spinner.getSelectedItem().toString();

        //get textView 引用
        TextView textView = findViewById(R.id.language);

        // 查询model
        String language = programExpert.getLanguage(feature);

        // set text
        textView.setText(language);
    }
}

File: Kotlin/ProgramAdviserK/app/src/main/java/bjfu/it/zhangsixuan/programadviserk/MainActivity.kt
package bjfu.it.zhangsixuan.programadviserk

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Spinner
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    private val programExpert = ProgramExpert()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    // click
    fun onClickButton(button: View){

        var spinner = findViewById<Spinner>(R.id.feature)

        var textView = findViewById<TextView>(R.id.language)

        var feature = spinner.selectedItem.toString()

        var language:String = programExpert.getLanguage(feature)

        textView.setText(language)

    }
}
